first_attempt_assign01

mugayamadox opened the pull request:
Check out my attempt to recreate the webpage in the assignment

most of the things seem fine but i have had issues with responsiveness.

mugayamadox committed:
first_attempt_assign01

Check out my attempt to recreate the webpage in the assignment

most of the things seem fine but i have had issues with responsiveness.

edwinmp commented:
Hey @mugayamadox, kudos on getting this done really quickly. I'll have a look asap. In the meantime, if you've installed the [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) extension on your VS Code, try to fix your code's formatting. It's important to write clean, maintainable code, especially when working in team. You have to put yourself in the shoes of your teammates and wonder if you'd like to look at the code you've written if someone else had written it. Fortunately, we have tools like Prettier that make this effortless and maintain consistency across dev teams.

As for responsiveness, great that you're getting a jump-start on that. It was outside the scope at this stage, but we'll cover it in the next bit.

edwinmp reviewed (changes requested):
Your implementation doesn't look like it does in the photo. There are some pretty clear differences. It's important to get it as close as possible since that's shows your understanding of the covered concepts.

edwinmp commented:
@mugayamadox [This](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/) is a decent guide to writing git commit messages.

mugayamadox committed:
style: improve formatting of code

style: write clean and maintainable code

This fixes the issues with format of the code in order to make it clean and maintainable

BREAKING CHANGE:
Before this fix, the code was appearing poorly formatted and was hard to read

Closes the first issue

edwinmp commented on apple-pie/styles/style.css:
```diff
@@ -21,4 +32,37 @@ body {
+h3 {
+  font-weight: bold;
+}
+#showcase {
+  background-image: url("../images/apple-pie.jpg");
+  height: 20em;
+}
+
+#content {
+  margin: 2em auto;
+  max-width: 30em;
+}
+
+#intro_text {
+  color: white;
+  font-size: 2em;
+  font-weight: bold;
+  padding: 5em 0em 1em 0em;
+  text-align: center;
+}
+
+#ingredients {
+  margin-top: 2em;
+}
+
+#directions {
+  margin-top: 2em;
+}
+
+img {
+  width: 300px;
+  max-height: 100px;
+}
```
You seem to like using IDs instead of classes. This is anti-pattern, and results in harder to maintain code since you'll likely have a lot of duplication. For example, instead of adding the same styles to `#directions` and `#instructions`, you can just create a single class that both elements use. [Here's ](https://www.youtube.com/watch?v=I_krGh8bzfk) a video that should explain it better.

mugayamadox replied:
well, i dont really have preferance for IDs over classes. the instructions to this project mentioned the use of IDs, so i thought maybe we are practising wih IDs first. I will be glad blending in the classes as recommended. thanks

mugayamadox committed:
Refactoring: fix issues with cover image, content  and image alignment

UPDATED:
Some div ids to classes for DRY.
The global "img" css rule to "img_content" to target only the images in content excluding the cover image.
"Directions" from paragraphs to an ordered list.

REMOVED:
center alignment of content

edwinmp reviewed (approved):
Great work. We can move on from this then. Will share next bits soon.

mugayamadox commented:
Thanks
I can't wait for the next bits.
Kind regards,
maddox

On Mon, Apr 8, 2024, 23:46 Edwin P. Magezi ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> Great work. We can move on from this then. Will share next bits soon.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/fourlanes-labs/lab-html-css-recipe/pull/1#pullrequestreview-1987377598>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIOPEDUORFFWTAIIKCVKMLTY4L6Y3AVCNFSM6AAAAABEREGL2OVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSOBXGM3TONJZHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
